### PR TITLE
:seedling: Make responseType configurable

### PR DESCRIPTION
**Pull request by jmelberg-okta**
### Description
Allows the OAuth 2.0 `response_type` to be configurable for implicit flows.

**Important:** `code` isn't explicitly supported yet.

Resolves: #109

### CODE REVIEW

File: packages/okta-angular/test/e2e/harness/src/app/app.module.ts [hunk 1]
```diff
@@ -48,7 +48,7 @@ const appRoutes: Routes = [
   },
   {
     path: 'sessionToken-login',
-    component: SessionTokenLogin
+    component: SessionTokenLoginComponent
```
Comment (robertjd) [inline packages/okta-angular/test/e2e/harness/src/app/app.module.ts]: This looks like a breaking change?

File: packages/okta-angular/test/e2e/harness/src/app/app.component.spec.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+import { environment } from './../environments/environment';
+
+import {
+  OktaAuthGuard,
+  OktaAuthModule,
+  OktaCallbackComponent,
+  OktaLoginRedirectComponent
+} from '@okta/okta-angular';
+
+describe('Unit Tests', () => {
```
Comment (robertjd) [inline packages/okta-angular/test/e2e/harness/src/app/app.component.spec.ts]: These spec files should be siblings of the source files, it's odd that we're burying unit testing within an e2e test.  We should discuss this, we need to decide if we're going to fix this now or later.  I also need a place to put unit testing for the user agent work

File: packages/okta-angular/src/okta/okta.config.ts [hunk 3]
```diff
@@ -18,6 +18,7 @@ export interface OktaConfig {
   clientId?: string;
   scope?: string;
   onAuthRequired?: Function;
+  responseType?: string;
```
Comment (robertjd) [inline packages/okta-angular/src/okta/okta.config.ts]: Should get some README love

File: packages/okta-angular/test/e2e/harness/src/app/app.module.ts (hunk 1 above)
Comment (jmelberg-okta) [reply packages/okta-angular/test/e2e/harness/src/app/app.module.ts]: This is used only in our test suite. We aren't exposing any of this via our SDK, so we aren't introducing a breaking change.

File: packages/okta-angular/README.md [hunk 4]
```diff
@@ -37,6 +37,7 @@ The `OktaAuthModule` is the initializer for your OpenID Connect client configura
   - `clientId` **(required)**: The OpenID Connect `client_id`
   - `redirectUri` **(required)**: Where the callback is hosted
   - `scope` *(optional)*: Reserved for custom claims to be returned in the tokens
+  - `responseType` *(optional)*: Desired token grant types
```
Comment (robertjd) [inline packages/okta-angular/README.md]: I feel like we should be explicitly mentioning that these are string values.

File: packages/okta-angular/test/e2e/harness/src/app/app.component.spec.ts (hunk 2 above)
Comment (robertjd) [reply packages/okta-angular/test/e2e/harness/src/app/app.component.spec.ts]: As discussed offline: this is going to be significant work and duplication of test boilerplate, so we won't do this.

File: packages/okta-angular/README.md (hunk 4 above)
Comment (jmelberg-okta) [reply packages/okta-angular/README.md]: I thought about noting that in this section, but seems weird to only mention the types on this configuration value. When this value is evaluated, typescript will error due to a type difference.

Comment (robertjd) [reply packages/okta-angular/README.md]: Aw, types 💘

**Pull request merged**